Cython/Utility/TypeConversion.c

```diff
@@ -276,37 +276,88 @@ static CYTHON_INLINE const char* __Pyx_PyUnicode_AsStringAndSize(PyObject* o, Py
 static CYTHON_INLINE const char* __Pyx_PyObject_AsStringAndSize(PyObject* o, Py_ssize_t *length) {
 #if __PYX_DEFAULT_STRING_ENCODING_IS_ASCII || __PYX_DEFAULT_STRING_ENCODING_IS_UTF8
     if (PyUnicode_Check(o)) {
-        return __Pyx_PyUnicode_AsStringAndSize(o, length);
+        const char* result = __Pyx_PyUnicode_AsStringAndSize(o, length);
+        if (unlikely(result == NULL)) {
+            goto error;
+        } else {
+            return result;
+        }
     } else
 #endif
 
     if (PyByteArray_Check(o)) {
+        int r = PyErr_WarnEx(
+            PyExc_RuntimeWarning,
+            "Cython: Pointer taken from resizable `bytearray` object. "
+            "To restrict resizing instead use `unsigned char[::1]`.",
```

Comment:
At first read, this seems to suggest using `::1` specifically to solve this problem, but that has nothing to do with it.
```suggestion
        int r = PyErr_WarnEx(
            PyExc_RuntimeWarning,
            "Cython: Pointer taken from resizable `bytearray` object. "
            "To restrict resizing instead use a memory view, e.g. `unsigned char[::1]`.",
```